iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/env/RegionMaintainHandler.java

```diff
@@ -693,6 +702,14 @@ public void transferRegionLeader(TConsensusGroupId regionId, TDataNodeLocation o
           (CONF.getSchemaRegionRatisRpcLeaderElectionTimeoutMaxMs()
                   + CONF.getSchemaRegionRatisRpcLeaderElectionTimeoutMinMs())
               / 2;
+      Integer leaderId = configManager.getLoadManager().getRegionLeaderMap().get(regionId);
+
+      if (leaderId != -1) {
+        // The migrated node is not leader, so we don't need to transfer temporarily
+        if (originalDataNode.getDataNodeId() != leaderId) {
+          return;
+        }
+      }
```

Comment:
1. `if (leaderId != -1)` useless check
2. inline leaderId

Reply:
I reconsidered here, maybe the check is necessary?

Reply:
leaderId may be -1, I think we should check it